src/main/java/com/mixpanel/android/mpmetrics/Tweaks.java

```diff
@@ -122,16 +122,46 @@ public synchronized boolean isNewValue(String tweakName, Object value) {
      * and will likely not be directly useful to code that imports the Mixpanel library.
      */
     public static class TweakValue {
-        private TweakValue(@TweakType int aType, Object aDefaultValue, Number aMin, Number aMax, Object value) {
+        private TweakValue(@TweakType int aType, Object aDefaultValue, Number aMin, Number aMax, Object value, String aName) {
             type = aType;
-            defaultValue = aDefaultValue;
+            name = aName;
             minimum = aMin;
             maximum = aMax;
+            if(minimum != null && maximum != null) {
+                try {
```

Comment:
Some notes in this piece of code:
- You don't need to cast that to a `Number` object and use primitives instead (you won't need a try/catch then)
- This could be a utility in this class or somewhere. Given a value, a min and a max, return the appropriate number so it's not out of bounds.
- This logic seems the same as doing `Math.min(Math.max(value, min), max);`